cmd/daprd/app/app.go

```diff
@@ -0,0 +1,179 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/automaxprocs/maxprocs"
+
+	// Register all components
+	_ "github.com/dapr/dapr/cmd/daprd/components"
+
+	"github.com/dapr/dapr/cmd/daprd/options"
+	"github.com/dapr/dapr/pkg/buildinfo"
+	bindingsLoader "github.com/dapr/dapr/pkg/components/bindings"
+	configurationLoader "github.com/dapr/dapr/pkg/components/configuration"
+	cryptoLoader "github.com/dapr/dapr/pkg/components/crypto"
+	lockLoader "github.com/dapr/dapr/pkg/components/lock"
+	httpMiddlewareLoader "github.com/dapr/dapr/pkg/components/middleware/http"
+	nrLoader "github.com/dapr/dapr/pkg/components/nameresolution"
+	pubsubLoader "github.com/dapr/dapr/pkg/components/pubsub"
+	secretstoresLoader "github.com/dapr/dapr/pkg/components/secretstores"
+	stateLoader "github.com/dapr/dapr/pkg/components/state"
+	workflowsLoader "github.com/dapr/dapr/pkg/components/workflows"
+	"github.com/dapr/dapr/pkg/modes"
+	"github.com/dapr/dapr/pkg/runtime/registry"
+	"github.com/dapr/dapr/pkg/security"
+	"github.com/dapr/kit/concurrency"
+	"github.com/dapr/kit/signals"
+
+	"github.com/dapr/dapr/pkg/runtime"
+	"github.com/dapr/kit/logger"
+)
+
+var (
+	log        = logger.NewLogger("dapr.runtime")
+	logContrib = logger.NewLogger("dapr.contrib")
+)
+
+func Run() {
+	// set GOMAXPROCS
+	_, _ = maxprocs.Set()
+
+	opts := options.New(os.Args[1:])
+
+	if opts.RuntimeVersion {
+		//nolint:forbidigo
+		fmt.Println(buildinfo.Version())
+		os.Exit(0)
+	}
+
+	if opts.BuildInfo {
+		//nolint:forbidigo
+		fmt.Printf("Version: %s\nGit Commit: %s\nGit Version: %s\n", buildinfo.Version(), buildinfo.Commit(), buildinfo.GitVersion())
+		os.Exit(0)
+	}
+
+	if opts.WaitCommand {
+		runtime.WaitUntilDaprOutboundReady(opts.DaprHTTPPort)
+		os.Exit(0)
+	}
+
+	// Apply options to all loggers.
+	opts.Logger.SetAppID(opts.AppID)
+
+	if err := logger.ApplyOptionsToLoggers(&opts.Logger); err != nil {
+		log.Fatal(err)
+	}
+
+	log.Infof("Starting Dapr Runtime -- version %s -- commit %s", buildinfo.Version(), buildinfo.Commit())
+	log.Infof("Log level set to: %s", opts.Logger.OutputLevel)
+
+	secretstoresLoader.DefaultRegistry.Logger = logContrib
+	stateLoader.DefaultRegistry.Logger = logContrib
+	cryptoLoader.DefaultRegistry.Logger = logContrib
+	configurationLoader.DefaultRegistry.Logger = logContrib
+	lockLoader.DefaultRegistry.Logger = logContrib
+	pubsubLoader.DefaultRegistry.Logger = logContrib
+	nrLoader.DefaultRegistry.Logger = logContrib
+	bindingsLoader.DefaultRegistry.Logger = logContrib
+	workflowsLoader.DefaultRegistry.Logger = logContrib
+	httpMiddlewareLoader.DefaultRegistry.Logger = log // Note this uses log on purpose
```

Comment:
I can answer that since I wrote that (it's not new).

I honestly don't know what the purpose is, except that it was that way before I touched that line. So I left the comment saying it was done on purpose, although I don't know what the original author's purpose was :)

Reply:
# Feb 28, 2020

When logging contrib was added, middleware did not have a logger as a parameter.  

- https://github.com/dapr/dapr/pull/1173
- https://github.com/dapr/dapr/blob/d75547e91d56ac4c2247b7156f575733fb21dea0/cmd/daprd/main.go#L285

# Feb 27, 2020

Middleware components were configured to allow for a logger to be passed in.  
Middleware logging passed in was in:  

- https://github.com/dapr/components-contrib/pull/236
- https://github.com/dapr/components-contrib/pull/238/files#diff-fe897f9d04c21756543bd4e1b9191d2492707eb8d3f3d592956aa966c24b6b97R33-R34

# Apr 8, 2020 

While fixing a bug, the new middleware logging parameter was used.  
Added middleware with logging as a parameter.  
https://github.com/dapr/dapr/pull/1379  

# The rest is history! Ever since then, middleware logging has been log and not contrib logging.

Three more middlewares registrations adding to the pattern and using the runtime logger.

- https://github.com/dapr/dapr/pull/2031
- https://github.com/dapr/dapr/pull/3180
- https://github.com/dapr/dapr/pull/1899

# Aug 25, 2022 ( Where ItalyPaleAle comes in )

Then... the main refactor... into todays form just keeping existing log setup in main:

-  Q: Still trying to workout if the work in pkg/components/middleware/http/registry.go::applyLogger was done to support keeping logging in the original configuration.
- https://github.com/dapr/dapr/pull/5062 

ItalyPaleAle added the note during a refactoring PR to at least highlight the difference introduced 2 years earlier.

- https://github.com/dapr/dapr/pull/5436

Reply:
TL;DR; Middleware is intentionally using the runtime logger or it was an accident. Definitely goes a few years back.